flink-table/flink-table-planner/src/test/java/org/apache/flink/table/catalog/CatalogFunctionTestBase.java

```diff
@@ -0,0 +1,140 @@
+
+	public static void setTableEnv(TableEnvironment e) {
+		tableEnv = e;
+	}
+
+	public abstract void execute() throws Exception;
+
+	@Test
+	public void testCreateCatalogFunctionInDefaultCatalog() {
+		String ddl1 = "create function f1 as 'org.apache.flink.function.TestFunction'";
+		tableEnv.sqlUpdate(ddl1);
+		assertTrue(Arrays.asList(tableEnv.listFunctions()).contains("f1"));
+
+		tableEnv.sqlUpdate("DROP FUNCTION IF EXISTS default_catalog.default_database.f1");
+		assertTrue(!Arrays.asList(tableEnv.listFunctions()).contains("f1"));
+	}
+
+	@Test
+	public void testCreateFunctionWithFullPath() {
+		String ddl1 = "create function default_catalog.default_database.f2 as" +
+			" 'org.apache.flink.function.TestFunction'";
+		tableEnv.sqlUpdate(ddl1);
+		assertTrue(Arrays.asList(tableEnv.listFunctions()).contains("f2"));
+
+		tableEnv.sqlUpdate("DROP FUNCTION IF EXISTS default_catalog.default_database.f2");
+		assertTrue(!Arrays.asList(tableEnv.listFunctions()).contains("f2"));
+	}
+
+	@Test
+	public void testCreateFunctionNotExists() {
+
+		String ddl1 = "create function catalog1.database1.f3 as 'org.apache.flink.function.TestFunction'";
+
+		try {
+			tableEnv.sqlUpdate(ddl1);
+		} catch (Exception e){
+			assertTrue(e.getMessage().equals("Catalog catalog1 does not exist"));
+		}
+	}
+
+	@Test
+	public void testDropFunctionNonExists() {
+		try {
+			tableEnv.sqlUpdate("DROP FUNCTION default_catalog.default_database.f4");
+		} catch (Exception e){
+			assertEquals(e.getMessage(), "Could not execute DROP FUNCTION:" +
+				" (identifier: [`default_catalog`.`default_database`.`f4`], IfExists: [false])");
+		}
+	}
+
+	@Test
+	public void testUseDefinedCatalogFunction() throws Exception {
+		List<Row> sourceData = Arrays.asList(
+			toRow(1, "1000", 2),
+			toRow(2, "1", 3),
+			toRow(3, "2000", 4),
+			toRow(1, "2", 2),
+			toRow(2, "3000", 3)
+		);
+
+		TestCollectionTableFactory.initData(sourceData, new ArrayList<Row>(), -1);
+
+		String sourceDDL = "create table t1(a int, b varchar, c int) with ('connector' = 'COLLECTION')";
+		String sinkDDL = "create table t2(a int, b varchar, c int) with ('connector' = 'COLLECTION')";
+
+		String functionDDL = "create function addOne as " +
+			"'org.apache.flink.table.catalog.CatalogFunctionTestBase$TestUDF'";
+
+		String query = " insert into t2 select t1.a, t1.b, addOne(t1.a, 1) as c from t1";
+
+		tableEnv.sqlUpdate(sourceDDL);
+		tableEnv.sqlUpdate(sinkDDL);
+		tableEnv.sqlUpdate(functionDDL);
+		tableEnv.sqlUpdate(query);
+		execute();
+
+		Row[] result = TestCollectionTableFactory.RESULT().toArray(new Row[0]);
+		Row[] expected = sourceData.toArray(new Row[0]);
+		assertArrayEquals(expected, result);
+	}
+
+	private Row toRow(Object ... objects) {
+		Row row = new Row(objects.length);
+		for (int i = 0; i < objects.length; i++) {
+			row.setField(i, objects[i]);
+		}
+
+		return row;
+	}
+
+	/**
+	 * Test udf class.
+	 */
+	public static class TestUDF extends ScalarFunction {
```

Comment:
We may also need to add test cases in blink-planner.

Reply:
Agree. Added.